src/golang/cmd/server/middleware/usage/usage.go

```diff
@@ -94,6 +93,23 @@ func reportUsage(startTime time.Time, environment string, statusCode int, urlPat
 }
 
 func WithUsageStats(environment string) func(http.Handler) http.Handler {
+	var machineID string
+	// Determine codespace ID based on
+	// https://docs.github.com/en/codespaces/developing-in-codespaces/default-environment-variables-for-your-codespace
+	if os.Getenv(codespacesEnv) == codespacesEnvActiveValue {
+		environment = "code_space"
+		machineID = os.Getenv(codespaceUserEnv)
+	} else {
+		// This call generates a unique hash of the host device in a privacy-preserving fashion.
+		// Details can be found here: https://github.com/denisbrodbeck/machineid
+		var err error
+		machineID, err = machineid.ProtectedID(hashKey)
```

Comment:
@cw75 I should be able to hit this line if I run `aqueduct clear` followed by `aqueduct start` right?

Reply:
If that's the case, as long as you can see `likawind` ID in our usage stats (in `code_space` env), we should be good to go

Reply:
Yeah I saw your id.